### PR TITLE
RIA-6509 Added handler to remove non detained data when switched to d…

### PR DESCRIPTION
**Pull request by ThomasKKC**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-6509

### Change description ###
Added handler to remove non detained data when switched to detained path

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: src/functionalTest/resources/templates/minimal-age-assessment-appeal-started.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "homeOfficeReferenceNumber": "123456789",
-  "homeOfficeDecisionDate": "{$TODAY}",
+  "decisionLetterReferenceNumber": "RP/50052/2022",
```
Comment (neha-aggarw) [inline src/functionalTest/resources/templates/minimal-age-assessment-appeal-started.json:2]: Ageassessment appeals start from AG. May be in next commit you can change the reference number to start with AG.

**Review by neha-aggarw — APPROVED** [review]

File: src/functionalTest/resources/scenarios/RIA-6509-change-non-detained-aaa-to-detained-aaa-edit-appeal.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    "uri": "/asylum/ccdAboutToSubmit",
+    "credentials": "LegalRepresentativeOrgSuccess",
+    "input": {
+      "eventId": "editAppeal",
+      "state": "appealStarted",
+      "caseData": {
+        "legalRepresentativeEmailAddress": "{TEST_LAW_FIRM_ORG_SUCCESS_USERNAME}",
+        "template": "minimal-age-assessment-appeal-started.json",
+        "replacements": {
+          "appellantInDetention": "Yes",
```
Comment (rajeshthuraiyur) [inline src/functionalTest/resources/scenarios/RIA-6509-change-non-detained-aaa-to-detained-aaa-edit-appeal.json:13]: this should be no as well

**Review by rajeshthuraiyur — APPROVED** [review]
LGTM

File: src/functionalTest/resources/scenarios/RIA-6509-change-non-detained-aaa-to-detained-aaa-edit-appeal.json (hunk 2 above)
Comment (neha-aggarw) [reply src/functionalTest/resources/scenarios/RIA-6509-change-non-detained-aaa-to-detained-aaa-edit-appeal.json:13]: This is the scenario where we started as non-detained appeal but in edit we mark it as detained. hence appellantInDetention = Yes in replacements.

**Review by neha-aggarw — APPROVED** [review]
LGTM

**Review by colmlynch60 — APPROVED** [review]
LGTM

**Pull request merged**